File: Java/chapter_10_challenge/app/src/main/java/com/glella/criminalintent/CrimeActivity.java
package com.glella.criminalintent;

import android.content.Context;
import android.content.Intent;
import android.support.v4.app.Fragment;

import java.util.UUID;

public class CrimeActivity extends SingleFragmentActivity {

    private static final String EXTRA_CRIME_ID = "com.gella.crime_id";

    @Override
    protected Fragment createFragment() {
        //return new CrimeFragment();
        UUID crimeID = (UUID) getIntent().getSerializableExtra(EXTRA_CRIME_ID);
        return  CrimeFragment.newInstance(crimeID);
    }

    public static Intent newIntent(Context packageContext, UUID crimeID) {
        Intent intent = new Intent(packageContext, CrimeActivity.class);
        intent.putExtra(EXTRA_CRIME_ID, crimeID);
        return intent;
    }
}

File: Kotlin/chapter_10/app/src/main/java/com/glella/criminalintent/CrimeActivity.kt
package com.glella.criminalintent

import android.content.Context
import android.content.Intent
import android.support.v4.app.Fragment
import java.util.*

class CrimeActivity : SingleFragmentActivity() {

    companion object {
        val EXTRA_CRIME_ID = "com.gella.crime_id"

        fun newIntent(packageContext: Context, crimeID: UUID): Intent {
            val intent = Intent(packageContext, CrimeActivity::class.java)
            intent.putExtra(EXTRA_CRIME_ID, crimeID)
            return intent
        }
    }


    override fun createFragment(): Fragment {
        //return CrimeFragment()
        val crimeID: UUID = intent.getSerializableExtra(EXTRA_CRIME_ID) as UUID
        return CrimeFragment.newInstance(crimeID)
    }
}
